src/app/state/effects/todos.effects.ts

```diff
@@ -72,8 +80,15 @@ export class TodosEffects {
             ofType(TODO_UPDATE_REQ),
             concatMap(({todo,completed})=>{
                 return this.todoService.updateTodo(todo,completed).pipe(
-                    map(()=>updateTodoSuccess({todo:todo, completed:completed})),
-                    catchError((err)=>[updateTodoFail({error:err})])
+                    map(()=>{
+                        this.toastr.success("Update todo succeed","Update todo");
```

Comment:
instead of use `this.toastr.*`, I will introduce an abstract of toast which would use toastr internally.